### PR TITLE
add expand_as op, test=develop

### PR DESCRIPTION
**Pull request by JesseyXujin**
add expand_as_op.cc , expand_as_op.cu , expand_as_op.h

### CODE REVIEW

File: paddle/fluid/operators/expand_as_op.cu [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+#include "paddle/fluid/operators/expand_as_op.h"
+
+namespace ops = paddle::operators;
+REGISTER_OP_CUDA_KERNEL(
+    expand_as, ops::ExpandAsKernel<paddle::platform::CUDADeviceContext, float>,
```
Comment (kuke) [inline paddle/fluid/operators/expand_as_op.cu:15]: Please add the registration of int64 & float16 compute kernel.

File: paddle/fluid/API.spec [hunk 2]
```diff
@@ -226,6 +226,7 @@ paddle.fluid.layers.sequence_enumerate (ArgSpec(args=['input', 'win_size', 'pad_
 paddle.fluid.layers.unique (ArgSpec(args=['x', 'dtype'], varargs=None, keywords=None, defaults=('int32',)), ('document', 'cab0b06e5683875f12f0efc62fa230a9'))
 paddle.fluid.layers.unique_with_counts (ArgSpec(args=['x', 'dtype'], varargs=None, keywords=None, defaults=('int32',)), ('document', '1cb59c65b41766116944b8ed1e6ad345'))
 paddle.fluid.layers.expand (ArgSpec(args=['x', 'expand_times', 'name'], varargs=None, keywords=None, defaults=(None,)), ('document', '33bc4f6010282ffe044d77be7ba7c275'))
+paddle.fluid.layers.expand_as (ArgSpec(args=['x', 'expand_tensor', 'name'], varargs=None, keywords=None, defaults=(None,)), ('document', '17a95c18f6bfdabbe1416f095aa8091c'))
```
Comment (heavengate) [inline paddle/fluid/API.spec:229]: expand_tensor改target_tensor会不会好些？

File: paddle/fluid/operators/expand_as_op.cc [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                   "Input(Out@GRAD) should not be null.");
+
+    auto x_dims = ctx->GetInputDim("X");
+    auto out_dims = ctx->GetInputDim(framework::GradVarName("Out"));
+    if (!ctx->IsRuntime() && x_dims[0] < 0) {
+      PADDLE_ENFORCE_EQ(
+          x_dims[0], out_dims[0],
+          "The first dimension size of Input(Out@GRAD) should be "
+          "equal to the crroresponding dimension size of Input(X)");
+    }
```
Comment (heavengate) [inline paddle/fluid/operators/expand_as_op.cc:94]: 这个是要判断啥

File: python/paddle/fluid/tests/unittests/test_expand_as_op.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (heavengate) [inline python/paddle/fluid/tests/unittests/test_expand_as_op.py:1]: 2018 -> 2019

**Pull request closed without merging**